File: java/operators/OperatorsAssignment.java
class OperatorsAssignment {
  public static void main(String[] args) {
    int a = 0, b = 0;

    System.out.println(String.format("a: %d", a));
    System.out.println(String.format("b: %d", b));

    System.out.println();

    a = 0;
    b = 3;

    System.out.println(String.format("a = 0; a: %d", a));
    System.out.println(String.format("b = 3; b: %d", b));

    System.out.println();

    a = b; // 3
    b = 5; // 5

    System.out.println(String.format("a = b; a: %d", a));
    System.out.println(String.format("b = 5; b: %d", b));

    System.out.println();

    a += 3; // 6
    System.out.println(String.format("a += 3; a: %d", a));

    a -= 2; // 4
    System.out.println(String.format("a -= 2; a: %d", a));

    a *= 2; // 8
    System.out.println(String.format("a *= 2; a: %d", a));

    a /= 4; // 2
    System.out.println(String.format("a /= 4; a: %d", a));

    a %= 3; // 2
    System.out.println(String.format("a %%= 3; a: %d", a));

    a <<= 2; // 8
    System.out.println(String.format("a <<= 2; a: %d", a));

    a >>= 1; // 4
    System.out.println(String.format("a >>= 1; a: %d", a));

    a &= 6; // 4
    System.out.println(String.format("a &= 6; a: %d", a));

    a |= 2; // 6
    System.out.println(String.format("a |= 2; a: %d", a));

    a ^= 3; // 5
    System.out.println(String.format("a ^= 3; a: %d", a));
  }
}

File: c#/operators/OperatorsAssignment.cs
class OperatorsAssignment {
  public static void Main(string[] args) {
    int a = 0, b = 0;

    System.Console.WriteLine(string.Format("a: {0}", a));
    System.Console.WriteLine(string.Format("b: {0}", b));

    System.Console.WriteLine();

    a = 0;
    b = 3;

    System.Console.WriteLine(string.Format("a = 0; a: {0}", a));
    System.Console.WriteLine(string.Format("b = 3; b: {0}", b));

    System.Console.WriteLine();

    a = b; // 3
    b = 5; // 5

    System.Console.WriteLine(string.Format("a = b; a: {0}", a));
    System.Console.WriteLine(string.Format("b = 5; b: {0}", b));

    System.Console.WriteLine();

    a += 3; // 6
    System.Console.WriteLine(string.Format("a += 3; a: {0}", a));

    a -= 2; // 4
    System.Console.WriteLine(string.Format("a -= 2; a: {0}", a));

    a *= 2; // 8
    System.Console.WriteLine(string.Format("a *= 2; a: {0}", a));

    a /= 4; // 2
    System.Console.WriteLine(string.Format("a /= 4; a: {0}", a));

    a %= 3; // 2
    System.Console.WriteLine(string.Format("a %= 3; a: {0}", a));

    a <<= 2; // 8
    System.Console.WriteLine(string.Format("a <<= 2; a: {0}", a));

    a >>= 1; // 4
    System.Console.WriteLine(string.Format("a >>= 1; a: {0}", a));

    a &= 6; // 4
    System.Console.WriteLine(string.Format("a &= 6; a: {0}", a));

    a |= 2; // 6
    System.Console.WriteLine(string.Format("a |= 2; a: {0}", a));

    a ^= 3; // 5
    System.Console.WriteLine(string.Format("a ^= 3; a: {0}", a));
  }
}
